src/components/helpers/pretty-text-input.js

```diff
@@ -531,10 +512,22 @@ export default createReactClass({
 
   removeCodeMirrorEditor: function() {
     const textBoxNode = this.textBoxRef;
+
     const cmNode = textBoxNode.firstChild;
     textBoxNode.removeChild(cmNode);
+
     this.codeMirror.off('change', this.onCodeMirrorChange);
     this.codeMirror.off('focus', this.onFocusCodeMirror);
+
+    if (this.debouncedOnChangeAndTagCodeMirror) {
+      // Cancel any trailing invocation
+      this.debouncedOnChangeAndTagCodeMirror.cancel();
+      this.debouncedOnChangeAndTagCodeMirror = null;
+
+      // Flush changes
+      this.onChange(this.codeMirror.getValue());
```

Comment:
@jdeal I think that we should flush changes when we are about to destroy the CM editor and cancel debounced events so we don't have to manage edge cases in `onBlur` or `onChangeAndTagCodeMirror`.

Reply:
Yep, I think it makes a lot more sense to do this here @ngryman!